{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyO2fZjd3FPN7nPdyzZQXARf",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/RodrigoEslava/Ms_project/blob/main/G_profiler_API.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install gprofiler-official"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "e_yZz895DecV",
        "outputId": "40a50fc7-a6c9-48a2-f7e7-fbb0ae9b4f35"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting gprofiler-official\n",
            "  Downloading gprofiler_official-1.0.0-py3-none-any.whl (9.3 kB)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.9/dist-packages (from gprofiler-official) (2.27.1)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.9/dist-packages (from requests->gprofiler-official) (1.26.15)\n",
            "Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.9/dist-packages (from requests->gprofiler-official) (2.0.12)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.9/dist-packages (from requests->gprofiler-official) (2022.12.7)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.9/dist-packages (from requests->gprofiler-official) (3.4)\n",
            "Installing collected packages: gprofiler-official\n",
            "Successfully installed gprofiler-official-1.0.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "syNKA7d5C_Pz"
      },
      "outputs": [],
      "source": [
        "from gprofiler import GProfiler\n",
        "import pandas as pd"
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "vYvbjlJrFr60"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "gp = GProfiler(return_dataframe=False)\n",
        "gp.profile(organism='hsapiens',\n",
        "            query=[\"10911\",\"54429\"])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bIPVZUZyD3LW",
        "outputId": "ff26d5dc-a4f5-4e62-c308-f74d50bb27c4"
      },
      "execution_count": 84,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[]"
            ]
          },
          "metadata": {},
          "execution_count": 84
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "result"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qJ-4Qfv8OMOO",
        "outputId": "ceefcbba-95ca-4aa7-fb55-e30a221eca06"
      },
      "execution_count": 82,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[]"
            ]
          },
          "metadata": {},
          "execution_count": 82
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "result[0][\"intersection_size\"]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "61LihtUHMcOW",
        "outputId": "262eade1-ea0a-4ef5-8468-3b1b636bfc4e"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "1"
            ]
          },
          "metadata": {},
          "execution_count": 19
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import gprofiler as gp\n",
        "\n",
        "# Criar função para converter int64 para int e float64 para float\n",
        "def converter_tipo(valor):\n",
        "    if isinstance(valor, np.int64):\n",
        "        return int(valor)\n",
        "    elif isinstance(valor, np.float64):\n",
        "        return float(valor)\n",
        "    else:\n",
        "        return valor\n",
        "\n",
        "# Ler o DataFrame\n",
        "df = pd.read_csv('predicted_edges_genes_less_known_pain_edges (1)', sep = \" \" )\n",
        "df.reset_index(drop=True)\n",
        "df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 424
        },
        "id": "JldxIGQqWSIU",
        "outputId": "4bae2276-a0fd-465f-98a3-65e542591d26"
      },
      "execution_count": 71,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         class_pred    prob_0    prob_1  gene_1  gene_2\n",
              "0                 1  0.016940  0.983060    1131   50838\n",
              "1                 1  0.018147  0.981853    1268    2790\n",
              "2                 1  0.018778  0.981222    1268    3358\n",
              "3                 1  0.018975  0.981025   10911   54429\n",
              "4                 1  0.019074  0.980926    1813    5731\n",
              "...             ...       ...       ...     ...     ...\n",
              "1024376           1  0.499999  0.500001    4804  135644\n",
              "1024377           1  0.499999  0.500001   10413    6786\n",
              "1024378           1  0.499999  0.500001    2643    1509\n",
              "1024379           1  0.500000  0.500000     409    3054\n",
              "1024380           1  0.500000  0.500000    4129    4597\n",
              "\n",
              "[1024381 rows x 5 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-5fd208a2-5b51-4623-82aa-e4b33cfc3423\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>class_pred</th>\n",
              "      <th>prob_0</th>\n",
              "      <th>prob_1</th>\n",
              "      <th>gene_1</th>\n",
              "      <th>gene_2</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>0.016940</td>\n",
              "      <td>0.983060</td>\n",
              "      <td>1131</td>\n",
              "      <td>50838</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>0.018147</td>\n",
              "      <td>0.981853</td>\n",
              "      <td>1268</td>\n",
              "      <td>2790</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>0.018778</td>\n",
              "      <td>0.981222</td>\n",
              "      <td>1268</td>\n",
              "      <td>3358</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>0.018975</td>\n",
              "      <td>0.981025</td>\n",
              "      <td>10911</td>\n",
              "      <td>54429</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>0.019074</td>\n",
              "      <td>0.980926</td>\n",
              "      <td>1813</td>\n",
              "      <td>5731</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024376</th>\n",
              "      <td>1</td>\n",
              "      <td>0.499999</td>\n",
              "      <td>0.500001</td>\n",
              "      <td>4804</td>\n",
              "      <td>135644</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024377</th>\n",
              "      <td>1</td>\n",
              "      <td>0.499999</td>\n",
              "      <td>0.500001</td>\n",
              "      <td>10413</td>\n",
              "      <td>6786</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024378</th>\n",
              "      <td>1</td>\n",
              "      <td>0.499999</td>\n",
              "      <td>0.500001</td>\n",
              "      <td>2643</td>\n",
              "      <td>1509</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024379</th>\n",
              "      <td>1</td>\n",
              "      <td>0.500000</td>\n",
              "      <td>0.500000</td>\n",
              "      <td>409</td>\n",
              "      <td>3054</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024380</th>\n",
              "      <td>1</td>\n",
              "      <td>0.500000</td>\n",
              "      <td>0.500000</td>\n",
              "      <td>4129</td>\n",
              "      <td>4597</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1024381 rows × 5 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5fd208a2-5b51-4623-82aa-e4b33cfc3423')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-5fd208a2-5b51-4623-82aa-e4b33cfc3423 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-5fd208a2-5b51-4623-82aa-e4b33cfc3423');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 71
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#selecionar as colunas gene 1 e gene 2\n",
        "gene_1_gene_2 = []\n",
        "gene_1_gene_2 = pd.DataFrame(gene_1_gene_2)\n",
        "gene_1_gene_2 = df[['gene_1', 'gene_2']]\n",
        "\n",
        "gene_1_gene_2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 424
        },
        "id": "YTgoaGxyZ_ol",
        "outputId": "c090b94e-2a3b-46db-e9a9-560a3f039ca1"
      },
      "execution_count": 72,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         gene_1  gene_2\n",
              "0          1131   50838\n",
              "1          1268    2790\n",
              "2          1268    3358\n",
              "3         10911   54429\n",
              "4          1813    5731\n",
              "...         ...     ...\n",
              "1024376    4804  135644\n",
              "1024377   10413    6786\n",
              "1024378    2643    1509\n",
              "1024379     409    3054\n",
              "1024380    4129    4597\n",
              "\n",
              "[1024381 rows x 2 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-7e2ece8c-0b57-4a45-97ac-7c2c1f5bc6a7\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>gene_1</th>\n",
              "      <th>gene_2</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1131</td>\n",
              "      <td>50838</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1268</td>\n",
              "      <td>2790</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1268</td>\n",
              "      <td>3358</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>10911</td>\n",
              "      <td>54429</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1813</td>\n",
              "      <td>5731</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024376</th>\n",
              "      <td>4804</td>\n",
              "      <td>135644</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024377</th>\n",
              "      <td>10413</td>\n",
              "      <td>6786</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024378</th>\n",
              "      <td>2643</td>\n",
              "      <td>1509</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024379</th>\n",
              "      <td>409</td>\n",
              "      <td>3054</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1024380</th>\n",
              "      <td>4129</td>\n",
              "      <td>4597</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1024381 rows × 2 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-7e2ece8c-0b57-4a45-97ac-7c2c1f5bc6a7')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-7e2ece8c-0b57-4a45-97ac-7c2c1f5bc6a7 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-7e2ece8c-0b57-4a45-97ac-7c2c1f5bc6a7');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 72
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df_top10 = []\n",
        "df_top10 = gene_1_gene_2[:10]\n",
        "\n",
        "df_top10"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 363
        },
        "id": "06cToqr-WyCB",
        "outputId": "2a8e826b-9382-42b7-d168-242f8cf8ca1f"
      },
      "execution_count": 73,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   gene_1  gene_2\n",
              "0    1131   50838\n",
              "1    1268    2790\n",
              "2    1268    3358\n",
              "3   10911   54429\n",
              "4    1813    5731\n",
              "5    5604    2676\n",
              "6    1131    3351\n",
              "7    1813   10886\n",
              "8    1129   10886\n",
              "9    3061   54429"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-4c62e675-0dc6-47fe-b133-6c260ce1989b\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>gene_1</th>\n",
              "      <th>gene_2</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1131</td>\n",
              "      <td>50838</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1268</td>\n",
              "      <td>2790</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1268</td>\n",
              "      <td>3358</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>10911</td>\n",
              "      <td>54429</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1813</td>\n",
              "      <td>5731</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>5604</td>\n",
              "      <td>2676</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>1131</td>\n",
              "      <td>3351</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>1813</td>\n",
              "      <td>10886</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>1129</td>\n",
              "      <td>10886</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>3061</td>\n",
              "      <td>54429</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-4c62e675-0dc6-47fe-b133-6c260ce1989b')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-4c62e675-0dc6-47fe-b133-6c260ce1989b button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-4c62e675-0dc6-47fe-b133-6c260ce1989b');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 73
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df_top10['gene_1'].to_list()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2vlaY_dviZav",
        "outputId": "89eac727-c659-4e79-a2d5-dbc1dd5d267e"
      },
      "execution_count": 67,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[1131, 1268, 1268, 10911, 1813, 5604, 1131, 1813, 1129, 3061]"
            ]
          },
          "metadata": {},
          "execution_count": 67
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from gprofiler import GProfiler\n",
        "\n",
        "# Criar uma lista para armazenar os resultados\n",
        "resultados = []\n",
        "gp = GProfiler(return_dataframe=False)\n",
        "# Iterar sobre cada linha do DataFrame\n",
        "for i, row in df_top10.iterrows():\n",
        "    # Realizar a query no GProfiler\n",
        "    res = gp.profile([converter_tipo(row['gene_1']), converter_tipo(row['gene_2'])], organism='hsapiens', user_threshold=0.05)\n",
        "    # Converter o resultado para um formato serializável em JSON\n",
        "    #res_json = json.loads(json.dumps(res, default=converter_tipo))\n",
        "\n",
        "    # Verificar se o resultado atende às condições\n",
        "    print(res)\n",
        "#    resultado = 0\n",
        "#    for r in res[0]:\n",
        "#      if 'source' in r and ('KEGG' in res['source'] or 'GO:BP' in res['source']) or 'REAC' in res['source'] and res['intersection_size'] == 2:\n",
        "#        resultado = 1\n",
        "#        break\n",
        "#   try:\n",
        "    resultado = 0\n",
        "    for r in res:\n",
        "      try:\n",
        "        if 'source' in r and ('KEGG' in r['source'] or 'GO:BP' in r['source']) or 'REAC' in r['source'] and r['intersection_size'] == 2:\n",
        "          resultado = 1\n",
        "          break\n",
        "        #if 'source' in res[0] and ('KEGG' in res[0]['source'] or 'BP' in res[0]['source']) and res[0]['intersection_size'] == 2:\n",
        "              #resultado = 1\n",
        "      except IndexError:\n",
        "           resultado = 0\n",
        "        #resultados.append(1)\n",
        "    else:\n",
        "        resultado = 0\n",
        "    # Adicionar a linha à lista de resultados\n",
        "    resultados.append(resultado)\n",
        "\n",
        "\n",
        "# Adicionar a lista de resultados ao DataFrame\n",
        "print(resultados)\n",
        "df_top10['Resultado'] = resultados\n",
        "\n",
        "# Exibir o DataFrame final\n",
        "print(df_top10)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OV-wOOmvWIBt",
        "outputId": "00a1d9d2-6208-4d27-9c69-2923c7c3944e"
      },
      "execution_count": 97,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[{'description': 'Taste transduction', 'effective_domain_size': 8091, 'intersection_size': 2, 'name': 'Taste transduction', 'native': 'KEGG:04742', 'p_value': 0.002754560421299043, 'parents': ['KEGG:00000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.023529411764705882, 'significant': True, 'source': 'KEGG', 'term_size': 85}, {'description': 'Absence of the abdominal musculature.', 'effective_domain_size': 4856, 'intersection_size': 1, 'name': 'Aplasia of the abdominal wall musculature', 'native': 'HP:0005199', 'p_value': 0.049983205642842504, 'parents': ['HP:0010318'], 'precision': 1.0, 'query': 'query_1', 'query_size': 1, 'recall': 1.0, 'significant': True, 'source': 'HP', 'term_size': 1}]\n",
            "[{'description': 'Retrograde endocannabinoid signaling', 'effective_domain_size': 8091, 'intersection_size': 2, 'name': 'Retrograde endocannabinoid signaling', 'native': 'KEGG:04723', 'p_value': 0.008393307636664162, 'parents': ['KEGG:00000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.013513513513513514, 'significant': True, 'source': 'KEGG', 'term_size': 148}, {'description': 'GNAI1-GNB3-GNG10 complex', 'effective_domain_size': 3383, 'intersection_size': 1, 'name': 'GNAI1-GNB3-GNG10 complex', 'native': 'CORUM:7078', 'p_value': 0.04993169344335535, 'parents': ['CORUM:0000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 1, 'recall': 0.5, 'significant': True, 'source': 'CORUM', 'term_size': 2}, {'description': 'GNAS-L-GNB3-GNG10 complex', 'effective_domain_size': 3383, 'intersection_size': 1, 'name': 'GNAS-L-GNB3-GNG10 complex', 'native': 'CORUM:7189', 'p_value': 0.04993169344335535, 'parents': ['CORUM:0000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 1, 'recall': 0.5, 'significant': True, 'source': 'CORUM', 'term_size': 2}]\n",
            "[{'description': '\"A G protein-coupled receptor signaling pathway in which the signal is transmitted via the activation or inhibition of a nucleotide cyclase activity and a subsequent change in the concentration of a cyclic nucleotide.\" [GOC:mah, GOC:signaling, ISBN:0815316194]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'G protein-coupled receptor signaling pathway, coupled to cyclic nucleotide second messenger', 'native': 'GO:0007187', 'p_value': 0.0044593514681632584, 'parents': ['GO:0007186'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.03225806451612903, 'significant': True, 'source': 'GO:BP', 'term_size': 62}, {'description': '\"Behavior associated with the intake of food.\" [GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'feeding behavior', 'native': 'GO:0007631', 'p_value': 0.013373338009494333, 'parents': ['GO:0007610'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.018691588785046728, 'significant': True, 'source': 'GO:BP', 'term_size': 107}, {'description': '\"Any process that modulates the rate, frequency, or extent of the regulated release of a peptide hormone from secretory granules.\" [GOC:tb]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'regulation of peptide hormone secretion', 'native': 'GO:0090276', 'p_value': 0.04101141268261606, 'parents': ['GO:0002791', 'GO:0030072', 'GO:0046883'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.0106951871657754, 'significant': True, 'source': 'GO:BP', 'term_size': 187}, {'description': 'GPCRs, class A rhodopsin-like', 'effective_domain_size': 7899, 'intersection_size': 2, 'name': 'GPCRs, class A rhodopsin-like', 'native': 'WP:WP455', 'p_value': 0.041519310127925896, 'parents': ['WP:000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.007782101167315175, 'significant': True, 'source': 'WP', 'term_size': 257}, {'description': '\"Any process that modulates the frequency, rate, or extent of peptide secretion.\" [GOC:add]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'regulation of peptide secretion', 'native': 'GO:0002791', 'p_value': 0.042341436071322694, 'parents': ['GO:0002790', 'GO:0051046', 'GO:0090087'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.010526315789473684, 'significant': True, 'source': 'GO:BP', 'term_size': 190}, {'description': '\"Any process that modulates the frequency, rate or extent of the directed movement of peptides, compounds of two or more amino acids where the alpha carboxyl group of one is bound to the alpha amino group of another, into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore.\" [GOC:dph, GOC:tb]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'regulation of peptide transport', 'native': 'GO:0090087', 'p_value': 0.04323990931794886, 'parents': ['GO:0015833', 'GO:0051049'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.010416666666666666, 'significant': True, 'source': 'GO:BP', 'term_size': 192}, {'description': '\"Any process that increases the concentration of calcium ions in the cytosol.\" [GOC:ai]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'positive regulation of cytosolic calcium ion concentration', 'native': 'GO:0007204', 'p_value': 0.04979334016407556, 'parents': ['GO:0065008'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.009708737864077669, 'significant': True, 'source': 'GO:BP', 'term_size': 206}]\n",
            "[]\n",
            "[{'description': '\"The series of molecular signals generated as a consequence of a dopamine receptor binding to one of its physiological ligands.\" [GOC:mah, PMID:21711983]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'dopamine receptor signaling pathway', 'native': 'GO:0007212', 'p_value': 0.0017474243458006152, 'parents': ['GO:0007165', 'GO:1903351'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.05128205128205128, 'significant': True, 'source': 'GO:BP', 'term_size': 39}, {'description': '\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a dopamine stimulus.\" [GO_REF:0000071, GOC:mr, GOC:TermGenie, PMID:11118945]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'cellular response to dopamine', 'native': 'GO:1903351', 'p_value': 0.008418765066812682, 'parents': ['GO:0071870', 'GO:1903350'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.023529411764705882, 'significant': True, 'source': 'GO:BP', 'term_size': 85}, {'description': '\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a dopamine stimulus.\" [GO_REF:0000071, GOC:mr, GOC:TermGenie, PMID:11118945]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'response to dopamine', 'native': 'GO:1903350', 'p_value': 0.00861921185411775, 'parents': ['GO:0071869'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.023255813953488372, 'significant': True, 'source': 'GO:BP', 'term_size': 86}, {'description': '\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a monoamine stimulus. A monoamine is any of a group of molecular messengers that contain one amino group that is connected to an aromatic ring by ethylene group (-CH2-CH2-). Monoamines are derived from the aromatic amino acids phenylalanine, tyrosine, histidine and tryptophan.\" [GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'cellular response to monoamine stimulus', 'native': 'GO:0071868', 'p_value': 0.011673077613647843, 'parents': ['GO:0071417', 'GO:0071867'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.02, 'significant': True, 'source': 'GO:BP', 'term_size': 100}, {'description': '\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a catecholamine stimulus. A catecholamine is any of a group of biogenic amines that includes 4-(2-aminoethyl)pyrocatechol [4-(2-aminoethyl)benzene-1,2-diol] and derivatives formed by substitution.\" [GOC:BHF, GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'cellular response to catecholamine stimulus', 'native': 'GO:0071870', 'p_value': 0.011673077613647843, 'parents': ['GO:0071407', 'GO:0071868', 'GO:0071869', 'GO:1901701'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.02, 'significant': True, 'source': 'GO:BP', 'term_size': 100}, {'description': '\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a monoamine stimulus. A monoamine is any of a group of molecular messengers that contain one amino group that is connected to an aromatic ring by ethylene group (-CH2-CH2-). Monoamines are derived from the aromatic amino acids phenylalanine, tyrosine, histidine and tryptophan.\" [GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'response to monoamine', 'native': 'GO:0071867', 'p_value': 0.012387611455452975, 'parents': ['GO:0010243'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.019417475728155338, 'significant': True, 'source': 'GO:BP', 'term_size': 103}, {'description': '\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a catecholamine stimulus. A catecholamine is any of a group of biogenic amines that includes 4-(2-aminoethyl)pyrocatechol [4-(2-aminoethyl)benzene-1,2-diol] and derivatives formed by substitution.\" [GOC:BHF, GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'response to catecholamine', 'native': 'GO:0071869', 'p_value': 0.012387611455452975, 'parents': ['GO:0014070', 'GO:0071867', 'GO:1901700'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.019417475728155338, 'significant': True, 'source': 'GO:BP', 'term_size': 103}, {'description': '\"A G protein-coupled receptor signaling pathway in which the signal is transmitted via the activation of adenylyl cyclase activity and a subsequent increase in the intracellular concentration of cyclic AMP (cAMP).\" [GOC:dph, GOC:mah, GOC:signaling, GOC:tb, ISBN:0815316194]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'adenylate cyclase-activating G protein-coupled receptor signaling pathway', 'native': 'GO:0007189', 'p_value': 0.028510607747273264, 'parents': ['GO:0007188'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.01282051282051282, 'significant': True, 'source': 'GO:BP', 'term_size': 156}, {'description': 'GPCRs, class A rhodopsin-like', 'effective_domain_size': 7899, 'intersection_size': 2, 'name': 'GPCRs, class A rhodopsin-like', 'native': 'WP:WP455', 'p_value': 0.041519310127925896, 'parents': ['WP:000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.007782101167315175, 'significant': True, 'source': 'WP', 'term_size': 257}, {'description': '\"Any process that increases the concentration of calcium ions in the cytosol.\" [GOC:ai]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'positive regulation of cytosolic calcium ion concentration', 'native': 'GO:0007204', 'p_value': 0.04979334016407556, 'parents': ['GO:0065008'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.009708737864077669, 'significant': True, 'source': 'GO:BP', 'term_size': 206}]\n",
            "[{'description': 'Ethanol metabolism production of ROS by CYP2E1', 'effective_domain_size': 7899, 'intersection_size': 1, 'name': 'Ethanol metabolism production of ROS by CYP2E1', 'native': 'WP:WP4269', 'p_value': 0.049973014572132435, 'parents': ['WP:000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 1, 'recall': 0.1111111111111111, 'significant': True, 'source': 'WP', 'term_size': 9}]\n",
            "[{'description': 'Monoamine GPCRs', 'effective_domain_size': 7899, 'intersection_size': 2, 'name': 'Monoamine GPCRs', 'native': 'WP:WP58', 'p_value': 0.0006664091606135952, 'parents': ['WP:000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.06060606060606061, 'significant': True, 'source': 'WP', 'term_size': 33}, {'description': '\"Combining with the biogenic amine serotonin and transmitting the signal across the membrane by activating an associated G-protein. Serotonin (5-hydroxytryptamine) is a neurotransmitter and hormone found in vertebrates and invertebrates.\" [GOC:ai]', 'effective_domain_size': 20171, 'intersection_size': 2, 'name': 'G protein-coupled serotonin receptor activity', 'native': 'GO:0004993', 'p_value': 0.0009979130590122775, 'parents': ['GO:0008227', 'GO:0099589'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.06896551724137931, 'significant': True, 'source': 'GO:MF', 'term_size': 29}, {'description': '\"The series of molecular signals generated as a consequence of a G protein-coupled serotonin receptor binding to one of its physiological ligands.\" [GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'G protein-coupled serotonin receptor signaling pathway', 'native': 'GO:0098664', 'p_value': 0.001025815911502386, 'parents': ['GO:0007186', 'GO:0007210'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.06666666666666667, 'significant': True, 'source': 'GO:BP', 'term_size': 30}, {'description': '\"Combining with the biogenic amine serotonin and transmitting a signal across a membrane by activating some effector activity. Serotonin (5-hydroxytryptamine) is a neurotransmitter and hormone found in vertebrates and invertebrates.\" [GOC:dos]', 'effective_domain_size': 20171, 'intersection_size': 2, 'name': 'serotonin receptor activity', 'native': 'GO:0099589', 'p_value': 0.0013788897194726276, 'parents': ['GO:0004888'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.058823529411764705, 'significant': True, 'source': 'GO:MF', 'term_size': 34}, {'description': '\"The series of molecular signals generated as a consequence of a serotonin receptor binding to one of its physiological ligands.\" [GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'serotonin receptor signaling pathway', 'native': 'GO:0007210', 'p_value': 0.0014031275111354472, 'parents': ['GO:0007165', 'GO:1903351'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.05714285714285714, 'significant': True, 'source': 'GO:BP', 'term_size': 35}, {'description': '\"Combining with an extracellular amine and transmitting the signal across the membrane by activating an associated G-protein; promotes the exchange of GDP for GTP on the alpha subunit of a heterotrimeric G-protein complex.\" [GOC:bf, GOC:dph]', 'effective_domain_size': 20171, 'intersection_size': 2, 'name': 'G protein-coupled amine receptor activity', 'native': 'GO:0008227', 'p_value': 0.0025439409262997714, 'parents': ['GO:0004930'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.043478260869565216, 'significant': True, 'source': 'GO:MF', 'term_size': 46}, {'description': 'Taste transduction', 'effective_domain_size': 8091, 'intersection_size': 2, 'name': 'Taste transduction', 'native': 'KEGG:04742', 'p_value': 0.002754560421299043, 'parents': ['KEGG:00000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.023529411764705882, 'significant': True, 'source': 'KEGG', 'term_size': 85}, {'description': 'Amine ligand-binding receptors', 'effective_domain_size': 10777, 'intersection_size': 2, 'name': 'Amine ligand-binding receptors', 'native': 'REAC:R-HSA-375280', 'p_value': 0.003985976571115242, 'parents': ['REAC:R-HSA-373076'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.047619047619047616, 'significant': True, 'source': 'REAC', 'term_size': 42}, {'description': '\"A G protein-coupled receptor signaling pathway in which the signal is transmitted via the activation or inhibition of a nucleotide cyclase activity and a subsequent change in the concentration of a cyclic nucleotide.\" [GOC:mah, GOC:signaling, ISBN:0815316194]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'G protein-coupled receptor signaling pathway, coupled to cyclic nucleotide second messenger', 'native': 'GO:0007187', 'p_value': 0.0044593514681632584, 'parents': ['GO:0007186'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.03225806451612903, 'significant': True, 'source': 'GO:BP', 'term_size': 62}, {'description': '\"A G protein-coupled receptor signaling pathway in which the signal is transmitted via the inhibition of adenylyl cyclase activity and a subsequent decrease in the intracellular concentration of cyclic AMP (cAMP).\" [GOC:dph, GOC:mah, GOC:signaling, GOC:tb, ISBN:0815316194]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'adenylate cyclase-inhibiting G protein-coupled receptor signaling pathway', 'native': 'GO:0007193', 'p_value': 0.007081667085613024, 'parents': ['GO:0007188'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.02564102564102564, 'significant': True, 'source': 'GO:BP', 'term_size': 78}, {'description': '\"A decrease in the diameter of blood vessels, especially arteries, due to constriction of smooth muscle cells that line the vessels, and usually causing an increase in blood pressure.\" [GOC:pr, ISBN:0192800752]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'vasoconstriction', 'native': 'GO:0042310', 'p_value': 0.007640559892569486, 'parents': ['GO:0097746'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.024691358024691357, 'significant': True, 'source': 'GO:BP', 'term_size': 81}, {'description': '\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a dopamine stimulus.\" [GO_REF:0000071, GOC:mr, GOC:TermGenie, PMID:11118945]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'cellular response to dopamine', 'native': 'GO:1903351', 'p_value': 0.008418765066812682, 'parents': ['GO:0071870', 'GO:1903350'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.023529411764705882, 'significant': True, 'source': 'GO:BP', 'term_size': 85}, {'description': '\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a dopamine stimulus.\" [GO_REF:0000071, GOC:mr, GOC:TermGenie, PMID:11118945]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'response to dopamine', 'native': 'GO:1903350', 'p_value': 0.00861921185411775, 'parents': ['GO:0071869'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.023255813953488372, 'significant': True, 'source': 'GO:BP', 'term_size': 86}, {'description': '\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a monoamine stimulus. A monoamine is any of a group of molecular messengers that contain one amino group that is connected to an aromatic ring by ethylene group (-CH2-CH2-). Monoamines are derived from the aromatic amino acids phenylalanine, tyrosine, histidine and tryptophan.\" [GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'cellular response to monoamine stimulus', 'native': 'GO:0071868', 'p_value': 0.011673077613647843, 'parents': ['GO:0071417', 'GO:0071867'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.02, 'significant': True, 'source': 'GO:BP', 'term_size': 100}, {'description': '\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a catecholamine stimulus. A catecholamine is any of a group of biogenic amines that includes 4-(2-aminoethyl)pyrocatechol [4-(2-aminoethyl)benzene-1,2-diol] and derivatives formed by substitution.\" [GOC:BHF, GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'cellular response to catecholamine stimulus', 'native': 'GO:0071870', 'p_value': 0.011673077613647843, 'parents': ['GO:0071407', 'GO:0071868', 'GO:0071869', 'GO:1901701'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.02, 'significant': True, 'source': 'GO:BP', 'term_size': 100}, {'description': '\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a monoamine stimulus. A monoamine is any of a group of molecular messengers that contain one amino group that is connected to an aromatic ring by ethylene group (-CH2-CH2-). Monoamines are derived from the aromatic amino acids phenylalanine, tyrosine, histidine and tryptophan.\" [GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'response to monoamine', 'native': 'GO:0071867', 'p_value': 0.012387611455452975, 'parents': ['GO:0010243'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.019417475728155338, 'significant': True, 'source': 'GO:BP', 'term_size': 103}, {'description': '\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a catecholamine stimulus. A catecholamine is any of a group of biogenic amines that includes 4-(2-aminoethyl)pyrocatechol [4-(2-aminoethyl)benzene-1,2-diol] and derivatives formed by substitution.\" [GOC:BHF, GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'response to catecholamine', 'native': 'GO:0071869', 'p_value': 0.012387611455452975, 'parents': ['GO:0014070', 'GO:0071867', 'GO:1901700'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.019417475728155338, 'significant': True, 'source': 'GO:BP', 'term_size': 103}, {'description': '\"Combining with a neurotransmitter and transmitting the signal to initiate a change in cell activity.\" [GOC:jl, GOC:signaling]', 'effective_domain_size': 20171, 'intersection_size': 2, 'name': 'neurotransmitter receptor activity', 'native': 'GO:0030594', 'p_value': 0.01393882994497197, 'parents': ['GO:0038023'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.018691588785046728, 'significant': True, 'source': 'GO:MF', 'term_size': 107}, {'description': '\"Any process that modulates the diameter of a tube.\" [GOC:bf]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'regulation of tube diameter', 'native': 'GO:0035296', 'p_value': 0.023942779194215457, 'parents': ['GO:0035150'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.013986013986013986, 'significant': True, 'source': 'GO:BP', 'term_size': 143}, {'description': '\"Any process that modulates the diameter of blood vessels.\" [GOC:pr]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'blood vessel diameter maintenance', 'native': 'GO:0097746', 'p_value': 0.023942779194215457, 'parents': ['GO:0003018', 'GO:0008015', 'GO:0035296'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.013986013986013986, 'significant': True, 'source': 'GO:BP', 'term_size': 143}, {'description': '\"Ensuring that a tube is of the correct length and diameter. Tube size must be maintained not only during tube formation, but also throughout development and in some physiological processes.\" [PMID:10887083]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'regulation of tube size', 'native': 'GO:0035150', 'p_value': 0.02428000143638757, 'parents': ['GO:0090066'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.013888888888888888, 'significant': True, 'source': 'GO:BP', 'term_size': 144}, {'description': '\"A synapse that uses serotonin as a neurotransmitter.\" [GOC:dos]', 'effective_domain_size': 22112, 'intersection_size': 1, 'name': 'serotonergic synapse', 'native': 'GO:0099154', 'p_value': 0.03325241906252891, 'parents': ['GO:0045202'], 'precision': 0.5, 'query': 'query_1', 'query_size': 2, 'recall': 0.5, 'significant': True, 'source': 'GO:CC', 'term_size': 2}, {'description': 'GPCRs, class A rhodopsin-like', 'effective_domain_size': 7899, 'intersection_size': 2, 'name': 'GPCRs, class A rhodopsin-like', 'native': 'WP:WP455', 'p_value': 0.041519310127925896, 'parents': ['WP:000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.007782101167315175, 'significant': True, 'source': 'WP', 'term_size': 257}, {'description': 'Absence of the abdominal musculature.', 'effective_domain_size': 4856, 'intersection_size': 1, 'name': 'Aplasia of the abdominal wall musculature', 'native': 'HP:0005199', 'p_value': 0.049983205642842504, 'parents': ['HP:0010318'], 'precision': 1.0, 'query': 'query_1', 'query_size': 1, 'recall': 1.0, 'significant': True, 'source': 'HP', 'term_size': 1}]\n",
            "[{'description': '\"Any process that modulates the frequency, rate or extent of adenylate cyclase activity.\" [GOC:go_curators]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'regulation of adenylate cyclase activity', 'native': 'GO:0045761', 'p_value': 0.0037636832063397954, 'parents': ['GO:0031279', 'GO:0051339'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.03508771929824561, 'significant': True, 'source': 'GO:BP', 'term_size': 57}, {'description': '\"Any process that modulates the frequency, rate or extent of lyase activity, the catalysis of the cleavage of C-C, C-O, C-N and other bonds by other means than by hydrolysis or oxidation, or conversely adding a group to a double bond. They differ from other enzymes in that two substrates are involved in one reaction direction, but only one in the other direction. When acting on the single substrate, a molecule is eliminated and this generates either a new double bond or a new ring.\" [GOC:ai]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'regulation of lyase activity', 'native': 'GO:0051339', 'p_value': 0.0067208628684639156, 'parents': ['GO:0050790'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.02631578947368421, 'significant': True, 'source': 'GO:BP', 'term_size': 76}, {'description': '\"Any process that modulates the frequency, rate or extent of cyclase activity.\" [GOC:mah]', 'effective_domain_size': 21128, 'intersection_size': 2, 'name': 'regulation of cyclase activity', 'native': 'GO:0031279', 'p_value': 0.0069000858782896045, 'parents': ['GO:0050790'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.025974025974025976, 'significant': True, 'source': 'GO:BP', 'term_size': 77}, {'description': 'GPCRs, class A rhodopsin-like', 'effective_domain_size': 7899, 'intersection_size': 2, 'name': 'GPCRs, class A rhodopsin-like', 'native': 'WP:WP455', 'p_value': 0.041519310127925896, 'parents': ['WP:000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.007782101167315175, 'significant': True, 'source': 'WP', 'term_size': 257}]\n",
            "[{'description': 'GPCRs, class A rhodopsin-like', 'effective_domain_size': 7899, 'intersection_size': 2, 'name': 'GPCRs, class A rhodopsin-like', 'native': 'WP:WP455', 'p_value': 0.041519310127925896, 'parents': ['WP:000000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 2, 'recall': 0.007782101167315175, 'significant': True, 'source': 'WP', 'term_size': 257}]\n",
            "[{'description': 'hypothalamus; synapses[≥Medium]', 'effective_domain_size': 11006, 'intersection_size': 1, 'name': 'hypothalamus; synapses[≥Medium]', 'native': 'HPA:0260152', 'p_value': 0.024961409660668416, 'parents': ['HPA:0260151'], 'precision': 1.0, 'query': 'query_1', 'query_size': 1, 'recall': 0.1, 'significant': True, 'source': 'HPA', 'term_size': 10}, {'description': 'hypothalamus; synapses[≥Low]', 'effective_domain_size': 11006, 'intersection_size': 1, 'name': 'hypothalamus; synapses[≥Low]', 'native': 'HPA:0260151', 'p_value': 0.0299536915928021, 'parents': ['HPA:0260000'], 'precision': 1.0, 'query': 'query_1', 'query_size': 1, 'recall': 0.08333333333333333, 'significant': True, 'source': 'HPA', 'term_size': 12}]\n",
            "[1, 1, 1, 0, 1, 0, 1, 1, 0, 0]\n",
            "   gene_1  gene_2  Resultado\n",
            "0    1131   50838          1\n",
            "1    1268    2790          1\n",
            "2    1268    3358          1\n",
            "3   10911   54429          0\n",
            "4    1813    5731          1\n",
            "5    5604    2676          0\n",
            "6    1131    3351          1\n",
            "7    1813   10886          1\n",
            "8    1129   10886          0\n",
            "9    3061   54429          0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "<ipython-input-97-75dd7145f487>:40: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  df_top10['Resultado'] = resultados\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import gprofiler as gp\n",
        "import json\n",
        "import numpy as np\n",
        "\n",
        "# Criar função para converter int64 para int e float64 para float\n",
        "def converter_tipo(valor):\n",
        "    if isinstance(valor, np.int64):\n",
        "        return int(valor)\n",
        "    elif isinstance(valor, np.float64):\n",
        "        return float(valor)\n",
        "    else:\n",
        "        return valor\n",
        "\n",
        "# Ler o DataFrame\n",
        "df = pd.read_csv('caminho/do/arquivo.csv')\n",
        "\n",
        "# Criar lista para armazenar os resultados\n",
        "resultados = []\n",
        "\n",
        "# Iterar sobre cada linha do DataFrame\n",
        "for i, row in df.iterrows():\n",
        "    # Realizar a query no GProfiler\n",
        "    res = gp.profile([converter_tipo(row['Coluna 1']), converter_tipo(row['Coluna 2'])], organism='hsapiens', user_threshold=0.05)\n",
        "    # Converter o resultado para um formato serializável em JSON\n",
        "    res_json = json.loads(json.dumps(res, default=converter_tipo))\n",
        "    # Verificar se o resultado atende às condições\n",
        "    if 'source' in res_json and ('KEGG' in res_json['source'] or 'BP' in res_json['source']) and res_json['intersection_size'] == 2:\n",
        "        resultado = 1\n",
        "    else:\n",
        "        resultado = 0\n",
        "    # Adicionar a linha à lista de resultados\n",
        "    resultados.append(resultado)\n",
        "\n",
        "# Adicionar a lista de resultados como uma nova coluna do DataFrame\n",
        "df['Resultado'] = resultados\n",
        "\n",
        "# Converter o DataFrame para JSON\n",
        "json_resultado = df.to_json(orient='records')\n",
        "\n",
        "# Exibir o JSON resultante\n",
        "print(json_resultado)\n"
      ],
      "metadata": {
        "id": "OonCX5-pl55F"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import gprofiler as gp\n",
        "import json\n",
        "import numpy as np\n",
        "\n",
        "# Criar função para converter int64 para int e float64 para float\n",
        "def converter_tipo(valor):\n",
        "    if isinstance(valor, np.int64):\n",
        "        return int(valor)\n",
        "    elif isinstance(valor, np.float64):\n",
        "        return float(valor)\n",
        "    else:\n",
        "        return valor\n",
        "\n",
        "# Ler o DataFrame\n",
        "df = pd.read_csv('caminho/do/arquivo.csv')\n",
        "\n",
        "# Criar lista para armazenar os resultados\n",
        "resultados = []\n",
        "\n",
        "# Iterar sobre cada linha do DataFrame\n",
        "for i, row in df.iterrows():\n",
        "    # Realizar a query no GProfiler\n",
        "    res = gp.profile([converter_tipo(row['Coluna 1']), converter_tipo(row['Coluna 2'])], organism='hsapiens', user_threshold=0.05)\n",
        "    \n",
        "    # Verificar se o resultado atende às condições\n",
        "    resultado = 0\n",
        "    for r in res:\n",
        "        if 'source' in r and ('KEGG' in r['source'] or 'BP' in r['source']) and r['intersection_size'] == 2:\n",
        "            resultado = 1\n",
        "            break\n",
        "\n",
        "    # Adicionar o resultado à lista de resultados\n",
        "    resultados.append(resultado)\n",
        "\n",
        "# Adicionar a lista de resultados como uma nova coluna do DataFrame\n",
        "df['Resultado'] = resultados\n",
        "\n",
        "# Converter o DataFrame para JSON\n",
        "json_resultado = df.to_json(orient='records')\n",
        "\n",
        "# Exibir o JSON resultante\n",
        "print(json_resultado)\n"
      ],
      "metadata": {
        "id": "pGHT9dXCnqZU"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import gprofiler as gp\n",
        "\n",
        "# Ler o DataFrame\n",
        "df = pd.read_csv('caminho/do/arquivo.csv')\n",
        "\n",
        "# Criar uma lista para armazenar os resultados\n",
        "resultados = []\n",
        "\n",
        "# Iterar sobre cada linha do DataFrame\n",
        "for i, row in df.iterrows():\n",
        "    # Realizar a query no GProfiler\n",
        "    res = gp.profile([row['Coluna 1'], row['Coluna 2']], organism='hsapiens', user_threshold=0.05)\n",
        "    # Verificar se o resultado atende às condições\n",
        "    if 'source' in res and ('KEGG' in res['source'] or 'BP' in res['source']) and res['intersection_size'] == 2:\n",
        "        resultados.append(1)\n",
        "    else:\n",
        "        resultados.append(0)\n",
        "\n",
        "# Adicionar a lista de resultados ao DataFrame\n",
        "df['Resultado'] = resultados\n",
        "\n",
        "# Exibir o DataFrame final\n",
        "print(df)\n"
      ],
      "metadata": {
        "id": "F9Tvt0xlr8q_"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}